unsplahAPI/unsplahAPI/Services/AppService/AppService.swift

```
@@ -6,3 +6,52 @@
 //
 
 import Foundation
+// TODO: Whole API logic
+class UnsplashAPI {
+    
+    static let shared = UnsplashAPI()
+    
+    private init() {}
+    // MARK: - Network
+    private let networkService: NetworkServiceProtocol = NetworkService()
+    
+    func askUnsplashAPIToExtractImages(page: Int, imageCollection: String?, imagesReceiverHandler: @escaping([UnsplashImage]) -> Void) {
+        let isTargetQuery = imageCollection != nil
+        let urlPath = self.linkBuilder(isTargetQuery: isTargetQuery, page: page, imageCollection: imageCollection)
+        networkService.requestPhotosFromUnsplash(urlPath: urlPath) { queryResponse in
+            switch queryResponse {
+            case .success(let data):
+                let images = self.decodeJsonObjectToArrayOfImages(data: data, isTargetQuery: isTargetQuery)
+                return imagesReceiverHandler(images)
+            case .failure(let error):
+                print(error)
```

we should return the completion in case of failure too, we can return the error or nil in this case

---

We need to notify the user if something went wrong, not print it for ourselves ))

---

> We need to notify the user if something went wrong, not print it for ourselves ))

so, the alert message will be enough for it or should I send him notification?

---

In Network Service you're handling the errors related to networking, but here you might have errors from parsing the data. Alert message is enough